platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -367,18 +373,27 @@ OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const C )
 {
     OtaPalStatus_t result = OTA_PAL_COMBINE_ERR( OtaPalUninitialized, 0 );
     char realFilePath[ OTA_FILE_PATH_LENGTH_MAX ];
+    char * pFileName = NULL;
 
     if( C != NULL )
     {
         if( C->pFilePath != NULL )
         {
             if( C->pFilePath[ 0 ] != ( uint8_t ) '/' )
             {
-                /* POSIX port using standard library */
-                /* coverity[misra_c_2012_rule_21_6_violation] */
-                int res = snprintf( realFilePath, OTA_FILE_PATH_LENGTH_MAX, "%s/%s", getenv( "PWD" ), C->pFilePath );
-                assert( res >= 0 );
-                ( void ) res; /* Suppress the unused variable warning when assert is off. */
+                /* Get current directory. */
+                pFileName = getcwd( realFilePath, OTA_FILE_PATH_LENGTH_MAX );
+
+                if( pFileName == NULL )
+                {
+                    LogError( ( "Failed to get current working directory: %s", strerror( errno ) ) );
+                }
+                else
+                {
+                    /* Add the filename . */
+                    realFilePath[ strlen( realFilePath ) ] = '/';
```

Comment:
This overwrites the `'\0'` at the end of `realFilePath` – it is no longer a valid NULL-terminated string.